fbw-a32nx/src/systems/instruments/src/EWD/index.tsx

```diff
@@ -1,34 +1,47 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { DisplayUnit } from '@instruments/common/displayUnit';
+import { useUpdate } from '@instruments/common/hooks';
 import { render } from '../Common';
 import UpperDisplay from './elements/UpperDisplay';
 import LowerLeftDisplay from './elements/LowerLeftDisplay';
 import LowerRightDisplay from './elements/LowerRightDisplay';
 import STS from './elements/STS';
 import ADV from './elements/ADV';
 import DownArrow from './elements/DownArrow';
-import PseudoFWC from './elements/PseudoFWC';
+import { NewPseudoFWC } from './NewPseudoFWC';
 
 import './style.scss';
 
-export const EWD: React.FC = () => (
-    <DisplayUnit
-        electricitySimvar="L:A32NX_ELEC_AC_ESS_BUS_IS_POWERED"
-        potentiometerIndex={92}
-    >
-        <svg className="ewd-svg" version="1.1" viewBox="0 0 768 768" xmlns="http://www.w3.org/2000/svg">
-            <UpperDisplay />
-            <line className="Separator" x1="4" y1="520" x2="444" y2="520" strokeLinecap="round" />
-            <line className="Separator" x1="522" y1="520" x2="764" y2="520" strokeLinecap="round" />
-            <line className="Separator" x1="484" y1="540" x2="484" y2="730" strokeLinecap="round" />
-            <LowerLeftDisplay x={14} y={554} />
-            <LowerRightDisplay x={520} y={554} />
-            <STS x={483} y={757} active={false} />
-            <ADV x={483} y={530} active={false} />
-            <DownArrow x={481} y={736} active={false} />
-            <PseudoFWC />
-        </svg>
-    </DisplayUnit>
-);
+export const EWD: React.FC = () => {
+    const [displayIndex] = useState(() => {
+        const url = document.getElementsByTagName('a32nx-ewd')[0].getAttribute('url');
+        return url ? parseInt(url.substring(url.length - 1), 10) : 0;
+    });
+
+    const [pseudoFwc] = useState(() => new NewPseudoFWC());
+
+    useUpdate((deltaTime) => pseudoFwc.onUpdate(deltaTime));
+
+    return (
+        <DisplayUnit
+            electricitySimvar="L:A32NX_ELEC_AC_2_BUS_IS_POWERED"
+            potentiometerIndex={displayIndex}
```

Comment:
DisplayIndex != potentionmeterIndex. I think the 92 before the change was correct

Reply:
Absolutely right, I have no idea how `displayIndex` even got involved in this. Fixed.